Assets/Scripts/SS3D/Core/Networking/PlayerControl/PlayerControlSystem.cs

```diff
@@ -0,0 +1,72 @@
+using System.Linq;
+using FishNet;
+using FishNet.Connection;
+using FishNet.Object;
+using FishNet.Object.Synchronizing;
+using SS3D.Core.Networking.PlayerControl.Messages;
+using SS3D.Core.Systems.Entities;
+using UnityEngine;
+
+namespace SS3D.Core.Networking.PlayerControl
+{
+    /// <summary>
+    /// Controls the player flux, when users want to authenticate, rejoin the game, leave the game
+    /// </summary>
+    public sealed class PlayerControlSystem : NetworkBehaviour
+    {
+        [SerializeField] private GameObject _soulPrefab;
+
+        [SyncObject]
+        private readonly SyncList<Soul> _serverSouls = new();
+
+        public string GetSoulCkeyByConn(NetworkConnection conn) => _serverSouls.Where(soul => soul.Owner == conn).ToArray()[0].Ckey;
```

Comment:
If you need to look up by an attribute, you should have a dictionary instead of a list.
In any case, `_serverSouls.SingleOrDefault(soul => soul.Owner == conn)?.Ckey` should be used instead.